### ISSUE TITLE
Wakelock requires a foreground activity

### ISSUE THREAD

**Issue by shinyford** [creativecreatorormaybenot/wakelock#23]
Hi - I'm consistently seeing this exception on Android:

```
Exception has occurred.
PlatformException (PlatformException(no_activity, wakelock requires a foreground activity, null))
```
As best I can tell, I have a foreground activity (if that means that the app is in the foreground at the point it happens).

Any idea why it's happening? Many thanks.

**Comment by creativecreatorormaybenot** [comment]
@shinyford This means that I have messed up something with the most recent change that I did not catch.

Please use version `0.1.3` while I attempt to fix this.

```yaml
wakelock: 0.1.3+4
```

Can you please send me your `flutter doctor`?

**Comment by pablanka** [comment]
I got the same issue. This is my `flutter doctor` output
```
[✓] Flutter (Channel master, v1.16.2-pre.28, on Mac OS X 10.15.3 19D76, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.43.1)
[✓] VS Code (version 1.26.1)
[✓] Connected device (3 available)

• No issues found!
```

**Comment by creativecreatorormaybenot** [comment]
@pablanka Thanks! That will do. I should be able to fix it (once I have figured out why I could not reproduce it).

I appreciate your feedback, it should also help to improve CI.

**Comment by creativecreatorormaybenot** [comment]
@pablanka @shinyford I actually think that I will not be able to fix it the way I thought.  
Potentially, I will need a more significant change if it is what I think (supporting the old plugin embedding).

Can you tell me if you are using the new Android APIs, i.e. your app is configured as described [here](https://flutter.dev/go/android-project-migration)?  
Not sure if this will help, but it might be useful to know the exact call to `Wakelock`, i.e. the full stack trace.

**Comment by kengu** [comment]
I had the same problem. Following the steps for upgrading pre 1.12 Android projects solved it as indicated below:
> Can you tell me if you are using the new Android APIs, i.e. your app is configured as described here?

**Comment by creativecreatorormaybenot** [comment]
@kengu Thank you, I think I have an idea how to solve this one.

**Comment by pablanka** [comment]
> @pablanka @shinyford I actually think that I will not be able to fix it the way I thought.
> Potentially, I will need a more significant change if it is what I think (supporting the old plugin embedding).
> 
> Can you tell me if you are using the new Android APIs, i.e. your app is configured as described [here](https://flutter.dev/go/android-project-migration)?

@creativecreatorormaybenot In my case, I've not updated the project yet. I'll do it and let you know.

**Comment by creativecreatorormaybenot** [comment]
@pablanka If possible, I would like to try to fix it for older projects and have someone test if it works.

**Comment by shinyford** [comment]
Sorry for being away - a lot has happened. Thanks!

> @pablanka @shinyford I actually think that I will not be able to fix it the way I thought.
> Potentially, I will need a more significant change if it is what I think (supporting the old plugin embedding).
> 
> Can you tell me if you are using the new Android APIs, i.e. your app is configured as described [here](https://flutter.dev/go/android-project-migration)?

I'm using the old bindings at the moment.

**Comment by creativecreatorormaybenot** [comment]
Could someone that experiences this problem in a project try to see if the following `wakelock` dependency fixes it?

```yml
wakelock:
  git:
    url: https://github.com/creativecreatorormaybenot/wakelock.git
    ref: 2bbce5f2a3d9c89ccdd9fd2d0ec8f0078e3cf7b4
```

**Comment by timolehto** [comment]
Trying that I'm getting this:
```
[        ] e: /home/timo/.pub-cache/git/wakelock-74c679cb450a13493b89bc42df1d8d0d895c95ce/android/src/main/kotlin/creativecreatorormaybenot/wakelock/WakelockPlugin.kt: (13, 14):
Redeclaration: WakelockPlugin
[   +1 ms] e: /home/timo/.pub-cache/hosted/pub.dartlang.org/wakelock-0.1.4/android/src/main/kotlin/creativecreatorormaybenot/wakelock/WakelockPlugin.kt: (13, 14): Redeclaration:
WakelockPlugin
[        ] e: /home/timo/.pub-cache/hosted/pub.dartlang.org/wakelock-0.1.4/android/src/main/kotlin/creativecreatorormaybenot/wakelock/WakelockPlugin.kt: (37, 14): Unresolved reference:
setup
[        ] FAILURE: Build failed with an exception.
[        ] * What went wrong:
[        ] Execution failed for task ':wakelock:compileDebugKotlin'.
[        ] > Compilation error. See log for more details
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 1s
```

**Comment by creativecreatorormaybenot** [comment]
@timolehto It looks like you have two versions of the plugin building at the same time.  
Make sure `wakelock` is only present once in your `pubspec.yaml`, i.e. only the `git` variant and not the version code.

Run `flutter clean` before running `flutter run` again.

**Comment by shinyford** [comment]
> Could someone that experiences this problem in a project try to see if the following `wakelock` dependency fixes it?
> 
> ```yaml
> wakelock:
>   git:
>     url: https://github.com/creativecreatorormaybenot/wakelock.git
>     ref: 74c679cb450a13493b89bc42df1d8d0d895c95ce
> ```
> 
> If so, I could upload it as the new version.

That works for me. No more error.

**Comment by creativecreatorormaybenot** [comment]
Thank you all for your collaboration 👍 Sadly, the Android integration tests have a different problem on GitHub Actions (I can only test it locally).

This issue should be fixed now and I am going to upload `0.1.4+1` now.

If you have any other problems with it, let me know!
